--- asterisk-1.2.17/apps/app_meetme.c   2007-04-05 20:35:28.000000000 -0400
+++ blindside/apps/app_meetme.c 2007-04-05 20:56:26.000000000 -0400
@@ -1409,9 +1409,28 @@
                                                                if (ztc.confmode & ZT_CONF_TALKER) {
                                                                        if (!ast_streamfile(chan, "conf-unmuted", chan->language))
                                                                                ast_waitstream(chan, "");
+
+                                                                        manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                                                "Channel: %s\r\n"
+                                                                                "Uniqueid: %s\r\n"
+                                                                                "Meetme: %s\r\n"
+                                                                                "Usernum: %i\r\n"
+                                                                                "Status: off\r\n",
+                                                                                chan->name, chan->uniqueid,
+                                                                                conf->confno, user->user_no);
+
                                                                } else {
                                                                        if (!ast_streamfile(chan, "conf-muted", chan->language))
                                                                                ast_waitstream(chan, "");
+
+                                                                        manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                                                "Channel: %s\r\n"
+                                                                                "Uniqueid: %s\r\n"
+                                                                                "Meetme: %s\r\n"
+                                                                                "Usernum: %i\r\n"
+                                                                                "Status: on\r\n",
+                                                                                chan->name, chan->uniqueid,
+                                                                                conf->confno, user->user_no);
                                                                }
                                                                break;
                                                        case '2': /* Un/Lock the Conference */
@@ -1489,9 +1508,27 @@
                                                                if (ztc.confmode & ZT_CONF_TALKER) {
                                                                        if (!ast_streamfile(chan, "conf-unmuted", chan->language))
                                                                                ast_waitstream(chan, "");
+
+                                                                        manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                                                "Channel: %s\r\n"
+                                                                                "Uniqueid: %s\r\n"
+                                                                                "Meetme: %s\r\n"
+                                                                                "Usernum: %i\r\n"
+                                                                                "Status: off\r\n",
+                                                                                chan->name, chan->uniqueid,
+                                                                                conf->confno, user->user_no);
                                                                } else {
                                                                        if (!ast_streamfile(chan, "conf-muted", chan->language))
                                                                                ast_waitstream(chan, "");
+
+                                                                        manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                                                "Channel: %s\r\n"
+                                                                                "Uniqueid: %s\r\n"
+                                                                                "Meetme: %s\r\n"
+                                                                                "Usernum: %i\r\n"
+                                                                                "Status: on\r\n",
+                                                                                chan->name, chan->uniqueid,
+                                                                                conf->confno, user->user_no);
                                                                }
                                                                break;
                                                        case '4':
@@ -2083,19 +2120,42 @@
                        case 77: /* M: Mute */
                                if (user) {
                                        user->adminflags |= ADMINFLAG_MUTED;
+                                        manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                "Channel: <unknown>\r\n"
+                                                "Uniqueid: <unknown>\r\n"
+                                                "Meetme: %s\r\n"
+                                                "Usernum: %i\r\n"
+                                                "Status: on\r\n",
+                                                cnf->confno, user->user_no);
                                } else {
                                        ast_log(LOG_NOTICE, "Specified User not found!\n");
                                }
                                break;
                        case 78: /* N: Mute all users */
                                AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
-                                       if (user && !(user->userflags & CONFFLAG_ADMIN))
+                                       if (user && !(user->userflags & CONFFLAG_ADMIN)) {
                                                user->adminflags |= ADMINFLAG_MUTED;
+
+                                               manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                       "Channel: <unknown>\r\n"
+                                                       "Uniqueid: <unknown>\r\n"
+                                                       "Meetme: %s\r\n"
+                                                       "Usernum: %i\r\n"
+                                                       "Status: on\r\n",
+                                                       cnf->confno, user->user_no);
+                                       }
                                }
                                break;
                        case 109: /* m: Unmute */
                                if (user && (user->adminflags & ADMINFLAG_MUTED)) {
                                        user->adminflags ^= ADMINFLAG_MUTED;
+                                        manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                "Channel: <unknown>\r\n"
+                                                "Uniqueid: <unknown>\r\n"
+                                                "Meetme: %s\r\n"
+                                                "Usernum: %i\r\n"
+                                                "Status: off\r\n",
+                                                cnf->confno, user->user_no);
                                } else {
                                        ast_log(LOG_NOTICE, "Specified User not found or he muted himself!\n");
                                }
@@ -2104,6 +2164,14 @@
                                AST_LIST_TRAVERSE(&cnf->userlist, user, list) {
                                        if (user && (user-> adminflags & ADMINFLAG_MUTED)) {
                                                user->adminflags ^= ADMINFLAG_MUTED;
+
+                                               manager_event(EVENT_FLAG_CALL, "MeetmeMute",
+                                                       "Channel: <unknown>\r\n"
+                                                       "Uniqueid: <unknown>\r\n"
+                                                       "Meetme: %s\r\n"
+                                                       "Usernum: %i\r\n"
+                                                       "Status: off\r\n",
+                                                       cnf->confno, user->user_no);
                                        }
                                }
                                break;

